### ISSUE TITLE
User uploads GeoJSON - user will see paths for the converted JSON, not their original GeoJSON

### ISSUE THREAD

**Issue by ghost** [Open-Telecoms-Data/cove-ofds#56]
eg the path's shown in additional checks errors section of UI.

Is this a problem? Do we need to make clear somehow?

**Comment by duncandewhurst** [comment]
Good catch. Ideally, we would show the path to the error in the GeoJSON file. If that's not possible, we should provide the necessary information (identifiers etc.) and guidance to help the user locate the error.

How is this issue handled in 360Giving (where data is usually uploaded in spreadsheet format) and in IATI (XML format)?

**Comment by duncandewhurst** [comment]
Noting that this issue interacts with #30

**Comment by duncandewhurst** [comment]
The 360Giving Data Quality tool explains that data is converted to JSON format for checking:

![image](https://user-images.githubusercontent.com/19265814/203460893-3cb5190e-440a-49fa-a299-abffebe17053.png)

But it also reports the sheet name, row number and column header to help users identify the location of the error in the spreadsheet:

![image](https://user-images.githubusercontent.com/19265814/203461048-f263e37d-c2d5-430c-85a7-bccde80f1c6c.png)

**Comment by ghost** [comment]
> How is this issue handled in 360Giving (where data is usually uploaded in spreadsheet format) and in IATI (XML format)?

Flatten tool already has built in functions that produce cell mappings. 

We'd have to code something similar into our GeoJSON->JSON conversion in our new lib. There may be some specs to work out first, such as for an org that is referenced multiple times do we just list the first time we see it or all times?

(We don't need it in our JSON->GeoJSON conversion right?)

Doable, but would take time - so I'll ask for prioritisation.

I guess a shortcut for now is  not show paths if GeoJSON was uploaded? Any path bits in the UI essentially need to be flexible anyway (cos ideally they would show CSV mappings too, so they have 3 possibilities!) so they will need work anyway.

**Comment by duncandewhurst** [comment]
> We'd have to code something similar into our GeoJSON->JSON conversion in our new lib. There may be some specs to work out first, such as for an org that is referenced multiple times do we just list the first time we see it or all times?

Yes, let's work that out in this issue.

> (We don't need it in our JSON->GeoJSON conversion right?)

Agreed.

> Doable, but would take time - so I'll ask for prioritisation.

I've added this issue and https://github.com/Open-Telecoms-Data/cove-ofds/issues/57 to the project board to-do column and prioritised them.

> I guess a shortcut for now is not show paths if GeoJSON was uploaded? Any path bits in the UI essentially need to be flexible anyway (cos ideally they would show CSV mappings too, so they have 3 possibilities!) so they will need work anyway.

Let's still show the paths until we have a better solution. I've made a PR to add a sentence to each results panel to clarify that errors are reported according to their location in the JSON format version of the data: https://github.com/Open-Telecoms-Data/cove-ofds/pull/59
